hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/namenode/ha/TestBootstrapStandby.java

```diff
@@ -172,6 +185,104 @@ public void testDownloadingLaterCheckpoint() throws Exception {
     restartNameNodesFromIndex(1);
   }
 
+  /**
+   * Test for downloading a checkpoint while the cluster is in rolling upgrade.
+   */
+  @Test
+  public void testRollingUpgradeBootstrapStandby() throws Exception {
```

Comment:
I added some comments to your draft change which would cause the test not to work. I will cherry-pick the tests to an older version to verify on my side that the test would fail before this change.

Reply:
Cool. Yeah, it would be great to verify the test fails outside of the `assertThrow` and getting a non-zero error code.

Reply:
I created a draft PR to show my minimal changes for the test. I added a comment with where the test fails and the output from the log.
https://github.com/apache/hadoop/pull/6031/files